modules/backing-up-red-hat-quay.adoc

```diff
@@ -0,0 +1,251 @@
+
+* A {productname} deployment on OpenShift Container Platform.
+
+
+.Procedure
+
+
+. Backup the `QuayRegistry` custom resource by exporting it:
++
+[source,terminal]
+----
+$ oc get quayregistry <quay-registry-name> -n <quay-namespace> -o yaml > quay-registry.yaml
+----
+
+. Edit the resulting `quayregistry.yaml` and remove the status section and the following metadata fields:
++
+[source,yaml]
+----
+  metadata.creationTimestamp
+  metadata.finalizers
+  metadata.generation
+  metadata.resourceVersion
+  metadata.uid
+----
+
+. Backup the managed keys secret:
++
+[NOTE]
+====
+If you are running a version older than Red Hat Quay 3.7.0, this step can be skipped. Some secrets are automatically generated while deploying Quay for the first time. These are stored in a secret called `<quay-registry-name>-quay-registry-managed-secret-keys` in the QuayRegistry namespace.
+====
++
+[source,terminal]
+----
+$ oc get secret -n <quay-namespace> <quay-registry-name>-quay-registry-managed-secret-keys -o yaml > managed-secret-keys.yaml
+----
+
+. Edit the the resulting `managed-secret-keys.yaml` file and remove all owner references. Your `managed-secret-keys.yaml` file should look similar to the following:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: <quayname>-quay-registry-managed-secret-keys
+  namespace: <quay-enterprise>
+data:
+  CONFIG_EDITOR_PW: <redacted>
+  DATABASE_SECRET_KEY: <redacted>
+  DB_ROOT_PW: <redacted>
+  DB_URI: <redacted>
+  SECRET_KEY: <redacted>
+  SECURITY_SCANNER_V4_PSK: <redacted>
+----
++
+All information under the `data` property should remain the same.
+
+. Backup the current Quay configuration:
++
+[source,terminal]
+----
+$ oc get secret -n <quay-namespace>  $(oc get quayregistry <quay-registry-name> -n <quay-namespace>  -o jsonpath='{.spec.configBundleSecret}') -o yaml > config-bundle.yaml
+----
+
+. Scale down the Quay component and edit the QuayRegistry custom resource by unmanaging the `horizontalpodautoscaler` component:
++
+[source,terminal]
+----
+$ oc patch QuayRegistry <quay-registry-name> -n <quay-namespace> -p '{"spec": {"components": [{"kind": "horizontalpodautoscaler", "managed": false }]}}' --type='merge'
+----
++
+Example YAML:
++
+[source,yaml]
+----
+  apiVersion: quay.redhat.com/v1
+  kind: QuayRegistry
+  metadata:
+    name: registry
+    namespace: ns
+  spec:
+    components:
+      …
+      - kind: horizontalpodautoscaler
+        managed: false
+      - kind: quay
+        managed: true
+        overrides:
+          replicas: 0
+      …
+----
++
+[NOTE]
+====
+If you are using a version older than 3.7.0, then you must manually scale down the Operator and manually scale down the Quay deployment. For example, you can run `$ oc scale --replicas=0 deployment quay-operator.v3.6.2 -n openshift-operators` to scale down the Operator, and `$ oc scale --replicas=0 deployment QUAY_MAIN_APP_DEPLOYMENT` to scale down the main application deployment.
+====
```

Comment:
If we are using 3.7.x,  do we not need to run the below commands?
```
$ oc scale --replicas=0 deployment quay-operator.v3.7.2 -n openshift-operators
$ oc scale --replicas=0 deployment QUAY_MAIN_APP_DEPLOYMENT
```
I checked the official document about `horizontalpodautoscaler`

```
horizontalpodautoscaler: Adjusts the number of Quay pods depending on memory/cpu consumption 
```
Does it have something with stopping quay pods during backup?

If the target is to stop quay pods during backup, no matter which quay version are using, should run 
```
 oc scale --replicas=0 deployment $(oc get deployment -n <quay-operator-namespace> |awk '/^quay-operator/ {print $1}') -n <quay-operator-namespace>
 oc scale --replicas=0 deployment $(oc get deployment -n <quay-namespace> -l quay-component=quay -o jsonpath='{.items[0].metadata.name}') -n <quay-namespace>
```

Reply:
@stevsmit Could you double confirm this part?  If my understanding is right,  we need to remove patch `horizontalpodautoscaler` part and `Note` part.  Just lead customer to run `oc scale --replicas=0 deployment.....` to stop quay pods during backup. 

The same change is needed to do with  backup environment recovery and quay restore part.